Src/BootCamp.Chapter/Inventory.cs

```diff
@@ -15,12 +15,29 @@ public Inventory()
 
         public Item[] GetItems(string name)
         {
-            return new Item[0];
+            Item[] wantedItems = new Item[0];
+            for (int i = 0; i < _items.Length; i++)
+            {
+                Item currentItem = _items[i];
+                if (currentItem.GetName() == name)
+                {
+                    wantedItems = Item.Append(wantedItems, currentItem);
+                }
+            }
+            return wantedItems;
```

Comment:
Highly inefficient!!

Reply:
Yes, because internally Append recreates arrays, so you create arrays on every iteration.
However at this stage I don't know if you can do it any better. So I guess it's fine @ellmaster what did you have in mind?

Reply:
Oh I think I see what you meant. indeed, there is something you can do to make this more efficient with the given knowledge.

Reply:
Yeah I wanted to point out that Item.Append is really bad performance-wise. There are multiple ways to tackle this with the current knowledge IMO @Almantask :)